.github/workflows/run-pytest.yml

```diff
@@ -0,0 +1,66 @@
+name: Run Unit Tests
+
+on:
+  # trigger on pull requests
+  pull_request:
+
+  # trigger on all commits to master
+  push:
+    branches:
+      - 'master'
+
+  # trigger on request
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        config: [ {python: '3.8', dependencies: 'newest'},
+                  {python: '3.9', dependencies: 'newest'},
+                  {python: '3.10', dependencies: 'newest'},
+                  {python: '3.10', dependencies: 'minimal'},
+                  {python: '3.8', dependencies: 'oldest'} ]
+        exclude:
+          # macOS, Python 3.10 is excluded because zarr dependency numcodecs is
+          # not yet available as a wheel for that configuration and fails to
+          # build from source. See
+          # https://github.com/zarr-developers/numcodecs/issues/326
+          - os: macos-latest
+            config: {python: '3.10', dependencies: 'newest'}
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.config.python }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.config.python }}
+    - name: Install newest dependencies
```

Comment:
You can use `actions/cache` to cache the files pip downloads and reduce the CI run time slightly: https://github.com/actions/cache

Reply:
I think this is a good idea but I'm going to leave it out-of-scope for this PR. Looks like this explains how to set it up for pip caching: https://github.com/actions/cache/blob/main/examples.md#simple-example

Reply:
Caching might not work well with cross-platform CI (on Windows in particular). 🤔

Reply:
Caching works fine on multi-platform CI. You include the platform and any other relevant system specific quantities (like python version) in the cache key.

Your "install newest" dependencies step takes ~3 m 18 s (out of 10m 56 s total test time). Caching would remove the download times from that 3m 18s, but not the install times.